sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -70,6 +76,57 @@ internal class NextLinkOperationImplementation : IOperation
             return new OperationToOperationOfT<T>(operationSource, operation);
         }
 
+        public static IOperation<T> Create<T>(
+            IOperationSource<T> operationSource,
+            IOperation operation)
+            => new OperationToOperationOfT<T>(operationSource, operation);
+
+        public static IOperation Create(
+            HttpPipeline pipeline,
+            RehydrationToken? rehydrationToken,
+            string? apiVersionOverride = null)
+        {
+            AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            var lroDetails = ModelReaderWriter.Write(rehydrationToken!, ModelReaderWriterOptions.Json).ToObjectFromJson<Dictionary<string, string>>();
+            if (!Uri.TryCreate(lroDetails["initialUri"], UriKind.Absolute, out var startRequestUri))
+            {
+                throw new ArgumentException("Invalid initial URI from rehydration token");
+            }
+
+            if (!lroDetails.TryGetValue("nextRequestUri", out var nextRequestUri))
+            {
+                throw new ArgumentException("Invalid next request URI from rehydration token");
+            }
+
+            RequestMethod requestMethod = new RequestMethod(lroDetails["requestMethod"]);
```

Comment:
Should any of the other dictionary lookups from rehydration token be validated?

Reply:
Construct a common method for this.